Unit.cpp

```diff
@@ -0,0 +1,44 @@
+#include "Unit.h"
+
+Unit::Unit(std::string name, int hp, int dmg)
+    : name(name), hp(hp), dmg(dmg)
+{
+}
+
+bool Unit::IsDead() const
+{
+    return !hp;
```

Comment:
hat a `return hp==0`-ra  vagy `return hp<=0`-ra gondoltam, ez kicsit hack, es kevesbe beszedes, de ok :-)

Reply:
Utobbi ugye akkor is mukodne, ha a logikaban nincs benne a 0-ra felhuzas lentebb. Igy egy kicsit "fugg" ez a fuiggveny a masiktol. Ha onnet kikerul ez a logika, akkor ez a fuggveny elromlik.